cmake/Configurations.cmake

```
@@ -20,73 +20,80 @@ include_guard(GLOBAL)
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINK_OPTIONS}" PARENT_SCOPE)
     endif()
-    if(ly_append_configurations_options_LINK_STATIC)
-        string(REPLACE ";" " " LINK_STATIC_STR "${ly_append_configurations_options_LINK_STATIC}")
-        string(APPEND LINK_STATIC_OPTIONS " " ${LINK_STATIC_STR})
-        set(LINK_STATIC_OPTIONS ${LINK_STATIC_OPTIONS} PARENT_SCOPE)
 
-        set(CMAKE_STATIC_LINKER_FLAGS ${LINK_STATIC_OPTIONS} PARENT_SCOPE)
+    if(ly_append_configurations_options_LINK_STATIC)
+        string(REPLACE ";" " " LINK_STR "${ly_append_configurations_options_LINK_STATIC}")
+        set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
     endif()
 
     if(ly_append_configurations_options_LINK_NON_STATIC)
-        string(REPLACE ";" " " LINK_NON_STATIC_STR "${ly_append_configurations_options_LINK_NON_STATIC}")
-        string(APPEND LINK_NON_STATIC_OPTIONS " " ${LINK_NON_STATIC_STR})
-        set(LINK_NON_STATIC_OPTIONS ${LINK_NON_STATIC_OPTIONS} PARENT_SCOPE)
+        string(REPLACE ";" " " LINK_STR "${ly_append_configurations_options_LINK_NON_STATIC}")
+        set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
+    endif()
 
-        set(CMAKE_MODULE_LINKER_FLAGS ${LINK_NON_STATIC_OPTIONS} PARENT_SCOPE)
-        set(CMAKE_SHARED_LINKER_FLAGS ${LINK_NON_STATIC_OPTIONS} PARENT_SCOPE)
-        set(CMAKE_EXE_LINKER_FLAGS ${LINK_NON_STATIC_OPTIONS} PARENT_SCOPE)
+    if(ly_append_configurations_options_LINK_EXE)
+        string(REPLACE ";" " " LINK_STR "${ly_append_configurations_options_LINK_EXE}")
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
     endif()
 
-    if(ly_append_configurations_options_LINK_EXECUTABLE)
-        string(REPLACE ";" " " LINK_EXECUTABLE_STR "${ly_append_configurations_options_LINK_EXECUTABLE}")
-        string(APPEND LINK_EXECUTABLE_OPTIONS " " ${LINK_EXECUTABLE_STR})
-        set(LINK_EXECUTABLE_OPTIONS ${LINK_EXECUTABLE_OPTIONS} PARENT_SCOPE)
+    if(ly_append_configurations_options_LINK_MODULE)
+        string(REPLACE ";" " " LINK_STR "${ly_append_configurations_options_LINK_MODULE}")
+        set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
+    endif()
 
-        set(CMAKE_EXE_LINKER_FLAGS ${LINK_EXECUTABLE_OPTIONS} PARENT_SCOPE)
+    if(ly_append_configurations_options_LINK_SHARED)
+        string(REPLACE ";" " " LINK_STR "${ly_append_configurations_options_LINK_SHARED}")
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_STR}" PARENT_SCOPE)
     endif()
 
     foreach(conf IN LISTS CMAKE_CONFIGURATION_TYPES)
```

Technically we could consolidate the pre-loop code with the for loop if we prepend an empty item to a list and iterator over that
```suggestion
    set(configuration_items ";${CMAKE_CONFIGURATION_TYPES}")
    foreach(conf IN LISTS configuration_items)
        unset(UCONF)
        if(conf)
           string(TOUPPER "_${conf}" UCONF)
        endif()
```

---

good idea, will do it in a different PR